### PR TITLE
Let docker assign random ports to the containers when running on CI

### PR DESCRIPTION
**Pull request by adrianclay**
When in development we want to expose the services to the host on a predictable port.  This allows for easier manual testing.

On CI this is unnecessary and can cause conflicts between two separate builds.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -4,7 +4,13 @@ APPS = asset-manager content-store govuk-content-schemas government-frontend \
 	collections frontend publisher calendars \
 	manuals-publisher manuals-frontend whitehall
 
-TEST_CMD = docker-compose run publishing-e2e-tests bundle exec rspec
+DOCKER_COMPOSE_CMD = docker-compose -f docker-compose.yml
+
+# Currently running in CI?
+ifndef ORIGIN_COMMIT
```
Comment (kevindew) [inline Makefile]: I think you want to go for a different env var here which is always set by jenkins. This one is only set when the tests are triggered by a different build (you'll see it wasn't used for the build for this)

Looking at the one set here on https://wiki.jenkins.io/display/JENKINS/Building+a+software+project maybe JENKINS_URL is the most explicit inidicator that you're on CI or we could set something in the Jenkinsfile.

**Comment by adrianclay** [comment]
> Development file seems like it might be useful for other things too. By specifying with -f will docker-compose.override.yml still get included if it's there? just wondering if that might be a place where we suggest people make local tweaks?

With the current state of the Makefile, override won't be loaded as we've specifically defined where ymls are.  We could add a

```
if override exists then
  ammend the docker_compose command
end if
```

**Comment by kevindew** [comment]
@adrianclay cool yeah that sounds a good idea to handle that. Guess it's not worth adding now unless we (well probably you) end up using the override for stuff currently.

Would commands like `docker-compose down` work still or would everything need arguments for the configs?

**Comment by adrianclay** [comment]
``` docker-compose down ``` would still work as normal, as that command will read the ` docker-compose.yml ` as previously which contains all the services information it needs to down the docker instances.

**Comment by kevindew** [comment]
Great!

**Pull request merged**